torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1888,6 +1888,23 @@ def wrapped_fn(x):
     return wrapped_fn
 
 
+def np_unary_ufunc_integer_promotion_wrapper_with_astype(fn):
+    # Check np_unary_ufunc_integer_promotion_wrapper
+    def is_integral(dtype):
+        return dtype in [np.bool_, bool, np.uint8, np.int8, np.int16, np.int32, np.int64]
+
+    # NOTE: Promotion in PyTorch is from integer types to the default dtype
+    np_dtype = torch_to_numpy_dtype_dict[torch.get_default_dtype()]
```

Comment:
Because the default dtype can change, the acquisition of the np_dtype should happen when the function is called, not when it's wrapped